boards/rapidsilicon/virgo_proto/doc/index.rst

```diff
@@ -0,0 +1,240 @@
++----------------+------------+----------------------+
+| DMA            | on-chip    | dma                  |
++----------------+------------+----------------------+
+| WATCHDOG       | on-chip    | wdt                  |
++----------------+------------+----------------------+
+| CRYPTO         | on-chip    | pufcc                |
++----------------+------------+----------------------+
+
+Other hardware features are not supported yet.
+
+Connections and IOs
+===================
+
+The Rapidsi Virgo platform has 1 GPIO controller. It providing 8 bits of IO.
+It is responsible for pin input/output, while the pull-up, etc are controlled
+by the pin controller module
+
+Mapping from GPIO controller to the Virgo Evaluation board pins:
+
++--------------------+--------------------+
+| GPIO controller    | Usage / Board pins |
++====================+====================+
+| **Push Buttons**   |                    |
++--------------------+--------------------+
+| GPIO.x             | -                  |
++--------------------+--------------------+
+
+Other peripheral mapping are listed below:
+
++-------------+---------------------------------+
+| Peripherals | Usage / Board pins              |
++=============+=================================+
+| SPI_1       | internal connected to SPI Flash |
++-------------+---------------------------------+
+
+System Clock
+------------
+
+The Rapidsi Virgo platform has 266MHz core clock.
+
+Serial Port
+-----------
+
+The Rapidsi Virgo platform has 1 UART.
+The Zephyr console output is by default assigned to UART0 and the default
+settings are 115200 8N1.
+
+Programming and debugging
+*************************
+
+For debugging zephyr applications or burning them into a flash, you will need to
+connect Rapidsi OpenOCD Port host computer to Virgo Evaluation board and execute the
+west flash command by filling in the appropriate IP and Port number in the board.cmake
+file.
+
+For Stepwise interactive debugging, follow the procedure mentioned below:
+
+Interactive Zephyr Flashing and Debugging
+===========================
+
+In order to do debugging of Zephyr RTOS over SoC prototype, follow the steps below:
+
+1 - In the project navigation pan of the VS Code, create a folder ‘.vscode’. Under this folder, create a file called launch.json as shown in picture below:
+
+.. image:: img/dbg_info/VSCode_WorkSpace.jpg
+     :align: center
+     :alt: VSCode Debug Launch Configuration
+
+2 - Run the OpenOCD or GDB Server as shown in the picture below:
+
+.. image:: img/dbg_info/OpenOCD_Start.jpg
+     :align: center
+     :alt: Run OpenOCD Server
+
+3 - In the launch.json, copy the following contents:
+
+.. code-block:: console
+   {
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "OpenOCD_JTAG_Connection",
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "/home/<users>/zephyrproject/zephyr/build/zephyr/zephyr.elf", // Path to your executable
```

Comment:
Perhaps better to use ZEPHYR_BASE here? Just a suggestion, if possible

Reply:
I actually tried this before pushing the changes in docs, but since launch configuration is not part of Zephyr itself, it does not recognise this environment variable. I will try if there is another way to get the path using the environment variable. If possible, i will definitely replace these path strings.

Reply:
I am marking the conversation closed. Change if you can, otherwise we are good